### PR TITLE
Add minimal pattern value length

### PR DESCRIPTION
**Pull request by meanrin**
Partial solution for #178 

**Changes:**

- add minimal len for pattern based values: 12
- move minimal len to config
- add min len argument to `ValueLengthCheck`
- add examples that detected as creds in current master, but not in this branch

**CredData benchmark:**

Master:
```
result_cnt : 4102, lost_cnt : 65, true_cnt : 3701, false_cnt : 336
credsweeper -> TP : 3701, FP : 336, TN : 19454363, FN : 882, FPR : 0.0000172709, FNR : 0.1924503600, ACC : 0.9999374078, PRC : 0.9167698786, RCL : 0.8075496400, F1 : 0.8587006961
```

This branch:
```
result_cnt : 4094, lost_cnt : 60, true_cnt : 3701, false_cnt : 333
credsweeper -> TP : 3701, FP : 333, TN : 19454366, FN : 882, FPR : 0.0000171167, FNR : 0.1924503600, ACC : 0.9999375619, PRC : 0.9174516609, RCL : 0.8075496400, F1 : 0.8589996519
```

Delta: 
```
FP : 336 -> 333
F1: 0.8587006961 -> 0.8589996519
```

Lines that detected by master, but removed in this branch:
```json
"line": "  \"refresh_token\": \"eyJrefresh\",",
"line_num": 214,
"path": "CredData/data/7738e44d/other/c.md",
"value": "eyJrefresh",

"line": "\t\t\t\t\tSo(authorizationHeaderCall1, ShouldStartWith, \"Bearer eyJ0e\")",
"line_num": 176,
"path": "CredData/data/8cda00f3/test/bu.go",
"value": "eyJ0e",
```

### CODE REVIEW

File: credsweeper/secret/config.json [hunk 1]
```diff
@@ -129,6 +129,8 @@
         ".jks"
     ],
     "check_for_literals": true,
+    "min_pattern_value_length": 12,
+    "min_keyword_value_length": 4,
```
Comment (babenek) [inline credsweeper/secret/config.json:133]: 4 symbols in base64 may encode 3 normal:
```
$ echo -n '{"a' | base64
eyJh
```
probably 6 is enough 
```
$ echo -n '{"a":' | base64
eyJhIjo=
```

**Review by babenek — COMMENTED** [review]
Could you add some samples to prove the solution?
From the issue like in #179

**Review by kmnls — DISMISSED** [review]

File: credsweeper/secret/config.json (hunk 1 above)
Comment (meanrin) [reply credsweeper/secret/config.json:133]: > 4 symbols in base64 may encode 3 normal:

@babenek 
I think question should be if 4/6 symbol json can realistically be a credential
'{"a":' is not even a valid json...

`echo -n '{0:0}' | base64` -> len 8
`echo -n '{"a":0}' | base64` -> len 12

So minimal length of 12 (`min_pattern_value_length`) would require payload to be at least not smaller than `{"a":0}`

Comment (meanrin) [reply credsweeper/secret/config.json:133]: @babenek 
Added negative examples with 35153123d922d557743fe91687d030c6575ff9b3

**Review by babenek — APPROVED** [review]
OK. 
Failed action test means previous version founds 19 items but current tests don't.
Let the PR will be merged with failed action check - it will be fixed after the merge.

**Review by csh519 — APPROVED** [review]
Thank you for adding minimal length value for filtering.

LGTM 👍

**Comment by babenek** [comment]
@meanrin https://github.com/Samsung/CredSweeper/pull/185 - suppose, minimal number of credentials is enough to check

**Comment by csh519** [comment]
Regarding `ActionTest`, it seems it could be passed by changing `.github/workflows/action.yml` file like below.

``` yml
    - name: Prepare commit status
      run: |
        ITEMS_CNT=$(jq '. | length' output.json)
        if [ 19 -eq ${ITEMS_CNT} ]; then
```

**Pull request merged**